src/timepicker/timepicker.spec.ts

```diff
@@ -45,722 +45,699 @@ function expectToDisplayTime(el: HTMLElement, time: string) {
+  fixture.detectChanges();
+
+  return fixture;
+}
+
 describe('ngb-timepicker', () => {
 
   beforeEach(() => {
-    TestBed.configureTestingModule({imports: [NgbTimepickerModule, FormsModule, ReactiveFormsModule]});
+    TestBed.configureTestingModule(
+        {declarations: [TestComponent], imports: [NgbTimepickerModule, FormsModule, ReactiveFormsModule]});
   });
 
   describe('rendering based on model', () => {
 
     // TODO: remove 'whenStable' once 'core/testing' is fixed
-    it('should render hour and minute inputs', async(inject([TestComponentBuilder], (tcb) => {
-         const html = `<ngb-timepicker [ngModel]="model"></ngb-timepicker>`;
-
-         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
-           fixture.componentInstance.model = {hour: 13, minute: 30};
-           fixture.detectChanges();
-           fixture.whenStable().then(() => {
-             fixture.detectChanges();
-             fixture.whenStable().then(() => { expectToDisplayTime(fixture.nativeElement, '13:30'); });
-           });
-         });
-       })));
+    it('should render hour and minute inputs', () => {
+      const html = `<ngb-timepicker [ngModel]="model"></ngb-timepicker>`;
+
+      const fixture = createTestComponent(html);
+      fixture.componentInstance.model = {hour: 13, minute: 30};
+      fixture.detectChanges();
+      fixture.whenStable().then(() => {
+        fixture.detectChanges();
+        fixture.whenStable().then(() => { expectToDisplayTime(fixture.nativeElement, '13:30'); });
+      });
+    });
 
     // TODO: remove 'whenStable' once 'core/testing' is fixed
-    it('should update inputs value on model change', async(inject([TestComponentBuilder], (tcb) => {
-         const html = `<ngb-timepicker [ngModel]="model"></ngb-timepicker>`;
-
-         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
-           fixture.componentInstance.model = {hour: 13, minute: 30};
-           fixture.detectChanges();
-           fixture.whenStable().then(() => {
-             fixture.detectChanges();
-             fixture.whenStable().then(() => {
-               expectToDisplayTime(fixture.nativeElement, '13:30');
-
-               fixture.componentInstance.model = {hour: 14, minute: 40};
-               fixture.detectChanges();
-               fixture.whenStable().then(() => {
-                 fixture.detectChanges();
-                 fixture.whenStable().then(() => { expectToDisplayTime(fixture.nativeElement, '14:40'); });
-               });
-             });
-           });
-         });
-       })));
+    it('should update inputs value on model change', () => {
+      const html = `<ngb-timepicker [ngModel]="model"></ngb-timepicker>`;
+
+      const fixture = createTestComponent(html);
+      fixture.componentInstance.model = {hour: 13, minute: 30};
+      fixture.detectChanges();
+      fixture.whenStable().then(() => {
```

Comment:
This pyramid of callbacks could be flattened with promise chaining. I did it in the typeahead test. That makes things easier to read IMHO, and reduces the indentation level.

```
    fixture.whenStable().then(() => {
        ...
        return fixture.whenStable();
    }).then(() => {
        ...
        return fixture.whenStable();
    }).then(() => {
        ...
    });
```